.github/workflows/ci.yaml

```diff
@@ -172,4 +172,16 @@ jobs:
         if: ${{ github.event_name != 'pull_request' }}
         run: |
           docker buildx imagetools inspect ghcr.io/${{ github.repository }}:${{ steps.meta.outputs.version }}
-
+  deploy:
+    needs: [ docker ]
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/main'
+    steps:
+    - name: deploying via ssh
+      uses: appleboy/ssh-action@master
+      with:
+        host: ${{ secrets.HOST }}
+        username: ${{ secrets.USERNAME }}
+        key: ${{ secrets.KEY }}
+        port: ${{ secrets.PORT }}
+        script: ./pull.sh
```

Comment:
Is that a script on the server? It should probably be checked in, in case we want to move onto a different server.

Reply:
I dont think this script is of interest in this case because you can actually change the deployment method.
So for example you can deploy it using docker or directly without containerization.

Reply:
Yea, should be fine. You can create your own script using the documentation anyway.

Reply:
I think the repo should also serve as a source of truth for our deployments. All relevant scripts and configs, except secrets, should be in the repo. This will make it easier to reason about changes and ensure easy reproduceability.